### ISSUE TITLE
[AGS4] Wrong(not precise) error in an "if" statement with an undefined variable.

### ISSUE THREAD

**Issue by tarnos12** [adventuregamestudio/ags#1839]
**Describe the bug**
```c
if(5 != myVariable)
```
There might be a bug in the compiler(?) or something which gives seemingly wrong error.
`Expected a term following the '!=', didn't find it`

Basically it can't find `myVariable` variable inside `if` statement, but the error doesn't specify that.

**AGS Version**
4.0(not latest)

**To Reproduce**
1. Create if statement with an operator `!=` where right hand side uses a variable that is not defined.
2. Run the project to see the error.
3. It seems to be the same for most other operators such as `==`, `<`, `>` etc.

**Expected behavior**
The error should be more consistent with similar message where left hand of the condition is not defined:
```c
if(myVariable != 1)
```
`Expected an expression, found 'myVariable' instead`

**Comment by fernewelten** [comment]
What happens whenever the compiler expects an expression is basically, 
- it starts scanning and stashing symbols that can be within expressions
- it continues this scanning until it finds a symbol that can't be in an expression (but doesn't consume that symbol yet)
- then it tries to make sense of what it has just stashed.

(As far as I know, both compilers do this.)

In this context, the compiler sees `if (` and starts scanning for an expression at this point. So 
- when you do `if (5 == myVariable)` then the compiler has stopped scanning for an expression in front of `myVariable`, because `myVariable` is an undefined symbol that can't be in an expression. Now it tries to make sense of `5 ==`. Whilst it's in the middle of processing this expression, it hits on a snafu: “There's a binary operator, but it doesn't have a right-hand side!”
- On the other hand, when you do `if (myVariable == 5)`, then the compiler has started scanning for an expression and has stopped immediately. It doesn't find any expression at all, __that's__ its snafu: “I need an expression at this point, but instead, there's that strange 'myVariable' symbol!”

So this explains why the cases are completely different from the viewpoint of the compiler, although they are similar from the viewpoint of the coder.

Taking the first case, it's kind of hard for the compiler to show `myVariable` in its error message, because the “expression” that it is currently processing doesn't __contain__ that symbol. The only thing that the compiler has at this point is “end of stashed symbols”. That's why the message simply says, “didn't find it”.

**Comment by ivan-mogilko** [comment]
> Expected an expression, found 'myVariable' instead

Would it be possible to somehow rephrase this error message, to help user understand that myVariable symbol is not known? Because this message reads like you are not supposed to use variable name there at all.

**Comment by ivan-mogilko** [comment]
Update: this problem still persists.

I.e. code like
```
  if (myVariable)
  {
  }
```

in case `myVariable` not declared, compiler reports:
```
Expected an expression, found 'myVariable' instead
```

Desired error message would be something like "undeclared identifier `myVariable`"

**Comment by ivan-mogilko** [comment]
Should be fixed by commit b598d5d7842d3af75afa5e27254521b39dfad24f (see #2516)
